### ISSUE TITLE
Research: Coherence Length ξ_C Symmetry Breaking Investigation

### ISSUE THREAD

**Issue by fermga** [fermga/TNFR-Python-Engine#2928]
# Coherence Length ξ_C Symmetry Breaking Investigation

## Context

The telemetry module (`src/tnfr/physics/fields.py`) provides coherence length ξ_C estimation via exponential decay fitting. Current evidence shows critical threshold behavior qualifying it as "weak-like, short-range" interaction regime with symmetry breaking at I_c.

**Current Status**: RESEARCH (NON-CANONICAL)

**From §10-11 evidence**:
- Critical threshold: I_c = 2.015 ± 0.005 (sharp phase transition)
- Coherence length: ξ_C ≈ 180-200 at stability
- Divergence behavior: +15% jump at fragmentation (critical slowing down)
- Topology dependence: ring (343) > ws (36) > scale-free (23)

## Objective

Investigate whether coherence length ξ_C exhibits predictive power comparable to Φ_s (|corr| > 0.5) and establish unique symmetry-breaking criteria analogous to weak interaction dynamics.

### 1. Critical Exponent Extraction (Correlation Length ν)

**Goal**: Determine if ξ_C diverges at I_c following power-law ξ_C ~ |I - I_c|^(-ν)

**Theory**: At continuous phase transitions, correlation length diverges with critical exponent ν:
- Mean-field: ν = 0.5
- 3D Ising: ν = 0.63
- 2D Ising: ν = 1.0

**Implementation**:
```python
def fit_correlation_length_exponent(data):
    """
    Fit ξ_C vs (I - I_c) to power-law.
    
    log(ξ_C) = log(A) - ν * log(|I - I_c|)
    """
    I_c = 2.015
    intensities = data['intensity']
    xi_c = data['coherence_length']
    
    # Filter near-critical points (avoid divergence noise)
    mask = np.abs(intensities - I_c) > 0.01
    x = np.log(np.abs(intensities[mask] - I_c))
    y = np.log(xi_c[mask])
    
    # Linear regression in log-log space
    nu, log_A = np.polyfit(x, y, 1)
    return -nu  # Negative because ξ ~ (I - I_c)^(-ν)
```

**Experiments**:
- [ ] Fine-grained intensity sweep: I ∈ [1.8, 2.0, 2.02, 2.03, 2.04, 2.05, 2.07, 2.1, 2.2]
- [ ] Measure ξ_C at each intensity (N=50+ runs for statistics)
- [ ] Fit power-law separately for I < I_c and I > I_c
- [ ] Compare ν_TNFR with theoretical predictions

**Hypothesis**: 
- If ν ≈ 0.5 → mean-field class (consistent with β = 0.556 from §12)
- If ν > 0.6 → lower-dimensional universality class

**Acceptance Criteria**:
- Power-law fit with R² > 0.8
- ν value determined with ±10% error
- Consistent across ≥3 topologies (universality)

### 2. Symmetry Breaking Mechanism

**Goal**: Identify what "symmetry" breaks at I_c and how ξ_C reflects this

**Theory**: Weak interaction analogy suggests **electroweak symmetry breaking**:
- Below I_c: High symmetry (long-range coherence, ξ_C ≈ N)
- At I_c: Critical point (symmetry breaking onset)
- Above I_c: Broken symmetry (fragmented domains, ξ_C ≈ fragment size)

**Phase Definition**:
```python
def measure_phase_symmetry(G):
    """
    Quantify phase field symmetry.
    
    High symmetry: Phases uniformly distributed or tightly clustered
    Low symmetry: Phases form distinct disconnected groups
    """
    phases = [G.nodes[n]['phase'] for n in G.nodes()]
    
    # Method 1: Circular variance (0 = perfect alignment, 1 = uniform)
    mean_phase = circmean(phases)
    circ_var = 1 - np.abs(np.mean(np.exp(1j * np.array(phases))))
    
    # Method 2: Cluster count (k-means on unit circle)
    from sklearn.cluster import KMeans
    phase_vectors = np.column_stack([np.cos(phases), np.sin(phases)])
    n_clusters = optimal_k_means(phase_vectors)  # Elbow method
    
    return circ_var, n_clusters
```

**Experiments**:
- [ ] Track circular variance vs intensity (I ∈ [1.5, 3.5])
- [ ] Measure cluster count evolution (does it jump at I_c?)
- [ ] Correlate with ξ_C: Does ξ_C ≈ avg(cluster size)?
- [ ] Test if fragmentation = formation of distinct phase domains

**Predicted Observations**:
- **I < I_c**: Low circular variance (aligned phases), 1-2 clusters, ξ_C ≈ N
- **I ≈ I_c**: Critical fluctuations, cluster count unstable, ξ_C diverges
- **I > I_c**: High circular variance (random phases), 5+ clusters, ξ_C ≈ cluster diameter

**Acceptance Criteria**:
- Circular variance jump > 50% crossing I_c
- Cluster count discontinuity at I_c (Δn ≥ 2)
- ξ_C ~ avg(cluster diameter) for I > I_c (R² > 0.7)

### 3. Screening Length Analogy

**Goal**: Test if ξ_C acts as screening length (weak interaction range)

**Theory**: In particle physics, weak interaction has characteristic length ~10^-18 m. Beyond this, interaction "screened" (exponentially suppressed).

**TNFR Interpretation**:
- **Screening length**: Distance beyond which phase coherence decays exponentially
- **Coherence function**: C(r) = C_0 * exp(-r / ξ_C)
- **Beyond ξ_C**: Nodes effectively decoupled (weak interaction range)

**Implementation**:
```python
def test_screening_hypothesis(G, source_node):
    """
    Measure coherence vs distance from source.
    
    If ξ_C = screening length, C(r) should decay exponentially.
    """
    coherence_by_distance = {}
    for node in G.nodes():
        d = nx.shortest_path_length(G, source_node, node)
        c_local = G.nodes[node].get('coherence', 0.5)  # Local coherence
        coherence_by_distance.setdefault(d, []).append(c_local)
    
    # Average coherence at each distance
    distances = sorted(coherence_by_distance.keys())
    avg_coherence = [np.mean(coherence_by_distance[d]) for d in distances]
    
    # Fit exponential decay
    def exp_decay(r, C0, xi):
        return C0 * np.exp(-r / xi)
    
    popt, _ = curve_fit(exp_decay, distances, avg_coherence, p0=[1.0, 50.0])
    xi_fitted = popt[1]
    
    return distances, avg_coherence, xi_fitted
```

**Experiments**:
- [ ] Select high-coherence seed nodes (top 10% by local C)
- [ ] Measure C(r) radial profile for each seed
- [ ] Fit exponential decay, extract ξ_fitted
- [ ] Compare ξ_fitted with ξ_C from BFS method (`fields.py`)
- [ ] Test if RA (resonance) extends ξ_C ("force carrier" mediation)

**Hypothesis**: 
- ξ_fitted ≈ ξ_C (±20% agreement) confirms screening interpretation
- RA-dominated sequences show ξ_C ↑ (longer interaction range)
- OZ-dominated sequences show ξ_C ↓ (shorter, localized disruption)

**Acceptance Criteria**:
- Exponential fit R² > 0.8 for ≥75% of seed nodes
- ξ_fitted vs ξ_C correlation > 0.7
- RA effect: Δξ_C ≥ +20% compared to baseline

### 4. Topology-Dependent Universality

**Goal**: Understand why ξ_C varies by topology despite universal β

**Observation** (from §9-10):
- Ring: ξ_C ≈ 343 (longest)
- WS: ξ_C ≈ 36 (intermediate)
- Scale-free: ξ_C ≈ 23 (shortest)
- Tree: ξ_C ≈ ? (to be measured)
- Grid: ξ_C ≈ ? (to be measured)

**Hypothesis**: ξ_C ~ f(topology) * ξ_intrinsic
- ξ_intrinsic: Universal coherence scale (phase transition)
- f(topology): Topology-dependent amplification factor

**Proposed Relationship**:
```python
def topology_coherence_factor(G):
    """
    Predict ξ_C from topology metrics.
    
    Candidates:
    - Diameter: max(shortest paths)
    - Avg path length: <d>
    - Clustering coefficient: C_g
    - Spectral gap: λ_2 (algebraic connectivity)
    """
    diameter = nx.diameter(G)
    avg_path = nx.average_shortest_path_length(G)
    clustering = nx.average_clustering(G)
    
    # Laplacian spectral gap
    L = nx.laplacian_matrix(G).toarray()
    eigenvalues = np.linalg.eigvalsh(L)
    spectral_gap = eigenvalues[1]  # Fiedler value
    
    return {
        'diameter': diameter,
        'avg_path': avg_path,
        'clustering': clustering,
        'spectral_gap': spectral_gap
    }
```

**Experiments**:
- [ ] Measure topology metrics for all 5 families
- [ ] Correlate each metric with observed ξ_C
- [ ] Fit regression: ξ_C = α * diameter + β * avg_path + ...
- [ ] Test predictive power on held-out topologies

**Expected Findings**:
- Ring: High diameter, low clustering → long ξ_C
- Scale-free: Low diameter (hubs), high clustering → short ξ_C
- Grid: Intermediate diameter, low clustering → intermediate ξ_C

**Acceptance Criteria**:
- Regression model R² > 0.8
- Dominant factor identified (diameter vs avg_path vs spectral_gap)
- Normalization: ξ_C / f(topology) ~ constant (universal intrinsic scale)

### 5. Early Warning System Development

**Goal**: Use ξ_C divergence as precursor signal for fragmentation

**Theory**: Near critical points, correlation length diverges **before** order parameter collapses.

**Implementation**:
```python
def monitor_coherence_length(G_trajectory):
    """
    Track ξ_C evolution over time.
    
    Early warning: ξ_C spike preceding C(t) drop.
    """
    xi_history = []
    c_history = []
    
    for G_t in G_trajectory:
        xi_c = estimate_coherence_length(G_t)
        c_t = compute_coherence(G_t)
        xi_history.append(xi_c)
        c_history.append(c_t)
    
    # Detect divergence: dξ/dt > threshold
    xi_gradient = np.gradient(xi_history)
    alert_steps = np.where(xi_gradient > threshold_gradient)[0]
    
    return xi_history, c_history, alert_steps
```

**Experiments**:
- [ ] Simulate trajectories approaching I_c from below
- [ ] Track ξ_C(t) and C(t) simultaneously
- [ ] Measure lead time: steps between ξ_C spike and C(t) drop
- [ ] Optimize threshold_gradient for <5% false positives

**Hypothesis**: ξ_C alerts 2-5 steps before C(t) < 0.3 (fragmentation)

**Acceptance Criteria**:
- Lead time ≥ 2 steps in ≥75% of fragmentation events
- False positive rate < 5% (stable trajectories not flagged)
- Complementary to Φ_s: Cases where ξ_C alerts but Φ_s doesn't (or vice versa)

### 6. Cross-Domain Validation

**Goal**: Test ξ_C symmetry breaking in biological/social/AI domains

**Proposed Domains**:
- **Biological**: Synchronization breakdown in neural networks (epileptic seizure as symmetry breaking)
- **Social**: Consensus collapse in opinion dynamics (polarization = domain formation)
- **AI**: Mode collapse in GANs (generator diversity loss = symmetry breaking)

**Implementation**:
- [ ] Define domain-specific coherence C(r) decay
- [ ] Measure ξ_C at stable vs critical states
- [ ] Test if critical transitions show ξ_C divergence + domain formation

**Acceptance Criteria**:
- ξ_C divergence observed in ≥2 domains at phase transitions
- Critical exponent ν within 30% of TNFR value (0.5 ± 0.15)
- Domain-specific interpretation documented

## Promotion Criteria to Canonical

From `fields.py` documentation:

1. **Predictive Power**: Demonstrate early warning capability (lead time ≥ 2 steps, <5% FP)
2. **Unique Safety Criteria**: ξ_C divergence alerts not captured by Φ_s drift
3. **Cross-Domain Validation**: Biological, social, AI applications

**Alternative Promotion Path**:
- Critical exponent ν measured with <10% error (universal constant)
- Screening length interpretation validated (R² > 0.8)
- Topology-dependent model with R² > 0.8 (predictive framework)

## Constraints

**Preserve Invariants**:
- **Invariant #1**: No direct EPI mutation (ξ_C is read-only telemetry)
- **Invariant #5**: Phase verification still required for UM/RA
- **Invariant #10**: Domain neutrality; "weak-like" analogy isolated to docs

**Classification**: RESEARCH until promotion criteria satisfied

## Tools & References

**Existing Tools**:
- `src/tnfr/physics/fields.py::estimate_coherence_length()`
- `benchmarks/u6_sequence_simulator.py`
- `tools/analyze_u6_results.py`

**Documentation**:
- `docs/TNFR_FORCES_EMERGENCE.md` §11 (critical threshold evidence)
- `AGENTS.md` § Structural Fields

**New Tools Needed**:
- Critical exponent fitter
- Phase symmetry analyzer
- Screening length tester
- Topology coherence predictor

## Expected Outcomes

**Best Case**:
- ν = 0.5 ± 0.05 (universal, mean-field class) → canonical as critical parameter
- Early warning system with 2-5 step lead time → unique safety criterion
- Screening length validated (R² > 0.8) → weak-like analogy empirically supported

**Realistic Case**:
- ν measured but noisy (0.4-0.6 range, ±20% error)
- Early warning works near I_c but unreliable far from criticality
- Topology model moderately predictive (R² ≈ 0.6-0.7)

**Worst Case**:
- ξ_C divergence too noisy near I_c (estimation method unreliable)
- No consistent lead time (varies 0-10 steps stochastically)
- Topology dependence too strong (no universal intrinsic scale)
- Remains RESEARCH telemetry

## Success Metrics

- [ ] Critical exponent ν measured with <10% error
- [ ] Symmetry breaking mechanism identified (circular variance jump > 50%)
- [ ] Screening length validated (R² > 0.8 exponential decay fit)
- [ ] Topology model predictive (R² > 0.8)
- [ ] Early warning lead time ≥ 2 steps, <5% FP
- [ ] Cross-domain validation on ≥2 domains

## Deliverables

1. **Analysis Report**: `docs/research/COHERENCE_LENGTH_VALIDATION.md`
2. **Code**: Critical exponent + screening tools in `src/tnfr/physics/fields.py`
3. **Tests**: `tests/physics/test_coherence_length_criticality.py`
4. **Data**: Experiment results in `benchmarks/results/coherence_length_*.jsonl`

---

**Estimated Effort**: 5-7 days of simulation + analysis
**Priority**: High (critical behavior already validated, divergence observed)
**Assigned to**: @copilot (automated research)
